core/src/main/java/com/graphhopper/util/InstructionList.java

```diff
@@ -417,9 +410,23 @@ public Instruction find(double lat, double lon, double maxDistance) {
         if (distCalc.calcDenormalizedDist(foundMinDistance) > maxDistance)
             return null;
 
-        // special case finish condition
-        if (foundInstruction == size())
-            foundInstruction--;
+        // Finish Instruction
+        if (foundPointIndex == getPoints().getSize() - 1)
+            return get(size() - 1);
+
+        int foundInstruction = -1;
+
+        for (int instructionIndex = 0; instructionIndex < size(); instructionIndex++) {
+            Instruction instruction = get(instructionIndex);
+            if (instruction.getFirst() <= foundPointIndex && instruction.getLast() > foundPointIndex) {
+                foundInstruction = instructionIndex;
+                break;
+            }
+        }
```

Comment:
Is this logic change of introducing two for-loops necessary or to make it easier?